src/gui/optionsdlg.cpp

```diff
@@ -425,6 +426,13 @@ OptionsDialog::OptionsDialog(QWidget *parent)
     m_ui->textTempPath->setDialogCaption(tr("Choose a save directory"));
     m_ui->textTempPath->setMode(FileSystemPathEdit::Mode::DirectorySave);
 
+    // disable mouse wheel event on widgets to avoid mis-selection
+    static Utils::Gui::WheelEventEater wheelEventEater;
```

Comment:
What's the point in making it `statis`, please?

Reply:
To prolong its lifetime, any suggestions?

Reply:
@Chocobo1, you can dynamically create it with options dialog as parent.

Reply:
Also you can use OptionsDialog as event filter. I don't think job this has such a general purpose to put it in separate class (especially inside Utils namespace).

Reply:
>@Chocobo1, you can dynamically create it with options dialog as parent.

OK.

> I don't think this job has such a general purpose to put it in separate class (especially inside Utils namespace).

I think it is convenient to have it, it is easier to filter wheel events for any widget.

Reply:
Is "filter wheel events" such a common task to pollute some global header with it?
Another thing is if it would be some generic class which accepts event type list on construction and filter them all...

Reply:
>Another thing is if it would be some generic class which accepts event type list on construction and filter them all...

Is there such use case in codebase?
If not I'll move WheelEventEater to optionsDialog.cpp

Reply:
>Is "filter wheel events" such a common task to pollute some global header with it?

It could be handy for dialogs which needs scrolling vertically, but there is no other such case than options dialog in qbt codebase currently.